PastriesDelivery/Managers/BusinessProviderManager.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System.Linq;
+
+namespace PastriesDelivery
+{
+    /// <summary>
+    /// This class contains methods intended for work with provider.
+    /// </summary>
+    public class BusinessProviderManager
+    {
+        private readonly IStorage _storage;
+
+        public BusinessProviderManager(IStorage storage)
+        {
+            _storage = storage;
+        }
+
+        public int SetId()
+        {
+            int id = default;
+
+            if (_storage.Products.Any())
+            {
+                return id = _storage.Products.Max(products => products.Pastry.Id) + 1;
```

Comment:
it can be just
return _storage.Products.Max(products => products.Pastry.Id) + 1;

and rename "products" to "product" cause it is single element